### ISSUE TITLE
GUID's not populating in `get_files_info()`

### ISSUE THREAD

**Issue by bgrich** [CenterForOpenScience/osfr#49]
I've noticed that the GUID column is not getting populated when calling `get_files_info()`. This is occurring for me with files that have had GUID's created for them by viewing directly on OSF. 

``` r
library(osfr)
get_files_info('m5pds')
#>                   name          materialized kind guid resource   provider
#> 1 public_test_file.csv /public_test_file.csv file   NA    m5pds osfstorage
#>               content_type                      created_utc
#> 1 application/octet-stream 2018-02-28T16:58:08.653632+00:00
#>                       modified_utc downloads version
#> 1 2018-02-28T16:58:08.653632+00:00        23       1
#>                                                                                    href
#> 1 https://files.osf.io/v1/resources/m5pds/providers/osfstorage/5a96dfa091b689000da32498
```

Whereas a direct call to the nodes endpoint on the API gives the GUID.

``` r
library(osfr)
osf_nodes_url <- 'https://api.osf.io/v2/nodes/m5pds/files/osfstorage/'
call <- httr::GET(osf_nodes_url)

res <- rjson::fromJSON(httr::content(call, 'text', encoding = "UTF-8"))

res$data[[1]]$attributes$guid
#> [1] "5z2bh"
```

The problem appears to be on WaterButler's end, as `get_files_info()` makes a direct call to WaterButler instead of going through the OSF API. This could be fixed by making a call to list all node files from the Nodes endpoint. The pagination will have to be dealt with, but it could be handled similarly to the already implemented search functions.

**Comment by chartgerink** [comment]
Yeah the Waterbutler - OSF API thing is very confusing. Will look into this when the time comes for this to be added/revised.

**Comment by bgrich** [comment]
I chatted with one of the devs on WaterButler and it sounds like it is an issue on their end. They may be able to fix the issue, but it probably won't be soon. At the very least, getting the information through the OSF API directly pulls down a GUID if it has been created.

**Comment by bgrich** [comment]
Addressed in PR #53.
